cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/proto/ReportContent.java

```diff
@@ -56,9 +56,18 @@ public long getTotal() {
   }
 
   /**
-   * @return the report records
+   * @return the records of program details in the report
    */
   public List<String> getDetails() {
     return details;
   }
+
+  public String toJson() {
+    return "{" +
+      "\"offset\":" + offset +
+      ", \"limit\":" + limit +
+      ", \"total\":" + total +
+      ", \"details\":" + details + // directly return details as JSON objects without stringifying them
+      '}';
```

Comment:
this looks weird, who do you want to do it this way ?

Reply:
If I just call Gson.toJson directly on class, since `details` is a list of String, it will be stringified. It's requested by UI that they should be returned as a list of JSON objects.